File: 백준/Silver/10819.차이를 최대로/차이를 최대로.java
import java.util.*;
import java.io.*;


public class Main {
    static int n;
    static int[] arr;
    static boolean[] visited;
    static int[] temp;
    static int result = 0;
    static void dfs(int depth){
        if(depth == n){
            int sum = 0;
            for(int i = 0; i < n-1; i++){
                sum += Math.abs(temp[i] - temp[i+1]);
            }

            result = Math.max(sum,result);
            return;
        }

        for(int i = 0; i < n; i++){
            if(!visited[i]){
                visited[i] = true;
                temp[depth] = arr[i];
                dfs(depth+1);
                visited[i] = false;
            }
        }
    }


    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        n = Integer.parseInt(br.readLine());
        arr = new int[n];
        visited = new boolean[n];
        temp = new int[n];

        StringTokenizer st = new StringTokenizer(br.readLine());
        for(int i = 0; i < n; i++){
            arr[i] = Integer.parseInt(st.nextToken());
        }

        dfs(0);
        System.out.println(result);

    }

}

File: 백준/Silver/10819. 차이를 최대로/차이를 최대로.py
import sys


def dfs(depth):
    if depth == n:
        global max_num
        temp = 0
        for i in range(n-1):
            temp += abs(result[i] - result[i+1])
            max_num = max(temp,max_num)
        return

    for i in range(0, n):
        if not visited[i]:
            visited[i] = True
            result[depth] = array[i]
            dfs(depth + 1)
            visited[i] = False


n = int(sys.stdin.readline())
array = list(map(int, sys.stdin.readline().split()))
visited = [False] * n
result = [0] * n
max_num = -1
dfs(0)

print(max_num)